### DISCUSSION TITLE
Fail to build on Linux due to 404 from mirror.bazel.build

### DISCUSSION THREAD

**Discussion by xuxinhang** [carbon-language/carbon-lang#1779] [Getting started] [answered]
I'm trying to build carbon-lang following the guide of README, but replace `brew` with `apt` on my Ubuntu 18.04 LTS. 

The `bazelisk` is downloaded from the latest GitHub release and added to PATH manually. llvm (and clang) are installed via `apt`. Soon after launching building, bazel complained unable to download packages, followed by errors issued:
    
    INFO: Invocation ID: 9849c062-82bd-4aba-b144-8d546424cfec
    WARNING: Download from https://mirror.bazel.build/ftp.gnu.org/gnu/m4/m4-1.4.18.tar.xz failed: class java.io.FileNotFoundException GET returned 404 Not Found
    WARNING: Download from https://mirror.bazel.build/github.com/jmillikin/rules_m4/releases/download/v0.1/m4-gnulib-788db09a9f88abbef73c97e8d7291c40455336d8.tar.xz failed: class java.io.FileNotFoundException GET returned 404 Not Found
    WARNING: Download from https://mirror.bazel.build/ftp.gnu.org/gnu/bison/bison-3.3.2.tar.xz failed: class java.io.FileNotFoundException GET returned 404 Not Found
    WARNING: Download from https://mirror.bazel.build/github.com/jmillikin/rules_bison/releases/download/v0.1/bison-gnulib-788db09a9f88abbef73c97e8d7291c40455336d8.tar.xz failed: class java.io.FileNotFoundException GET returned 404 Not Found
    INFO: Analyzed target //explorer:explorer (0 packages loaded, 0 targets configured).
    INFO: Found 1 target...
    ERROR: /home/li/.cache/bazel/_bazel_li/0228db5d83e20d51b81b9fb458bad627/external/m4_v1.4.18/gnulib/BUILD.bazel:193:11: Compiling gnulib/lib/sig-handler.c [for host] failed: undeclared inclusion(s) in rule '@m4_v1.4.18//gnulib:gnulib':
    this rule is missing dependency declarations for the following files included by 'gnulib/lib/sig-handler.c':
      '/usr/lib/llvm-6.0/lib/clang/6.0.0/include/stddef.h'
    ERROR: /home/li/.cache/bazel/_bazel_li/0228db5d83e20d51b81b9fb458bad627/external/m4_v1.4.18/gnulib/BUILD.bazel:193:11: Compiling gnulib/lib/secure_getenv.c [for host] failed: undeclared inclusion(s) in rule '@m4_v1.4.18//gnulib:gnulib':
    this rule is missing dependency declarations for the following files included by 'gnulib/lib/secure_getenv.c':
      '/usr/lib/llvm-6.0/lib/clang/6.0.0/include/stddef.h'
    ERROR: /home/li/.cache/bazel/_bazel_li/0228db5d83e20d51b81b9fb458bad627/external/m4_v1.4.18/gnulib/BUILD.bazel:193:11: Compiling gnulib/lib/fpending.c [for host] failed: undeclared inclusion(s) in rule '@m4_v1.4.18//gnulib:gnulib':
    this rule is missing dependency declarations for the following files included by 'gnulib/lib/fpending.c':
      '/usr/lib/llvm-6.0/lib/clang/6.0.0/include/stddef.h'
      '/usr/lib/llvm-6.0/lib/clang/6.0.0/include/stdarg.h'
    Target //explorer:explorer failed to build
    Use --verbose_failures to see the command lines of failed build steps.
    INFO: 7 processes: 7 internal.
    FAILED: Build did NOT complete successfully

The URLs given by the log does return HTTP 404.

How can I solve this problem? Should I switch to another Bazel version?

**Comment by jonmeow** [comment, marked as answer]
The mirror issues are _warnings_, not _errors_. The first error is here:

```
ERROR: /home/li/.cache/bazel/_bazel_li/0228db5d83e20d51b81b9fb458bad627/external/m4_v1.4.18/gnulib/BUILD.bazel:193:11: Compiling gnulib/lib/sig-handler.c [for host] failed: undeclared inclusion(s) in rule '@m4_v1.4.18//gnulib:gnulib':
this rule is missing dependency declarations for the following files included by 'gnulib/lib/sig-handler.c':
  '/usr/lib/llvm-6.0/lib/clang/6.0.0/include/stddef.h'
```

From this, I can observe that you're using a system install of LLVM v6. This is old, and also doesn't match our install advice.

Please see: https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/contribution_tools.md#troubleshooting-build-issues
